lib/ibmiconnector.js

```diff
@@ -1209,7 +1372,9 @@ exports.initialize = (ds, cb) => {
   dataSource.driver = odbc; // Provide access to the native driver
   dataSource.connector = new IBMiConnector(dataSource.settings);
   dataSource.connector.dataSource = ds;
-  dataSource.connector.pool.init((error) => {
+  odbc.pool(dataSource.connector.connectionString, (error, pool) => {
```

Comment:
As explained in https://github.com/strongloop/loopback-connector-ibmi/issues/7#issuecomment-565018318, I think this call should go to connector's `connect` method and this line should be changed as follows:

```js
  if (cb) {
    if (ds.settings.lazyConnect) {
      process.nextTick(cb);
    } else {
      ds.connecting = true;
      ds.connector.connect(callback);
    }
  }
```

Personally, I would try the following version to see it if works, I find it easier to understand:

```js
if (ds.settings.lazyConnect) {
  if (cb) process.nextTick(cb);
} else {
  ds.connect(cb);
}
```